### PR TITLE
Allow multiple other names in AstroSource

### LINKED ISSUE #34: Allow several alternative names for astrosources.
Sometimes there have been errors or different conventions in source names.
We should rename other_name to other_names and allow it to be a comma separated list of names.
It will allow IOP4 to work with the different conventions used along the history of -several- programs within one catalog.

E.g. 2145+067 a.k.a B 2145+067  was wrongly named B2135 in the catalog, but was observed with that name during some time due to an error in the observers spreadsheet. We could include the wrong name in the catalog (and a indication that the name is only because of an error, to reduce the files with that name).

### PR DESCRIPTION
**Pull request by juanep97**
Closes #34

### CODE REVIEW

File: iop4lib/instruments/instrument.py [hunk 1]
```diff
@@ -183,13 +183,14 @@ def _get_header_hintobject_hard(self, rawfit):
         """Get a hint for the AstroSource in this image from the header. 
         Return None if none found.
 
-        This method tries to match the OBJECT keyword with the possible names 
-        of each sources in the DB.
+        This method tries to match the OBJECT keyword with each source in the
+        DB, first tryin all of them by their main names, then by their other
```
Comment (morcuended) [inline iop4lib/instruments/instrument.py:187]:
```suggestion
        DB, first trying all of them by their main names, then by their other 
```

File: iop4lib/instruments/dipol.py [hunk 2]
```diff
@@ -272,14 +272,14 @@ def get_invariable_str(s):
...
             for source in catalog:
-                if not source['other_name']:
+                if not source.other_names_list:
                     continue
-                if get_invariable_str(search_str) in get_invariable_str(source['other_name']):
-                    return AstroSource.objects.get(name=source['name'])
-
+                if any([get_invariable_str(search_str) in get_invariable_str(other_name) for other_name in source.other_names_list]):
+                    return source
+
```
Comment (morcuended) [inline iop4lib/instruments/dipol.py:279]:
```suggestion
 
```

File: tests/conftest.py [hunk 3]
```diff
@@ -30,7 +30,7 @@
 import hashlib
 from pathlib import Path
 
-TESTDATA_MD5SUM = '4d393377f8c659e2ead2fa252a9a38b2'
+TESTDATA_MD5SUM = '29e4c96eaf6edc811c84aafd9e4110d0'
```
Comment (morcuended) [inline tests/conftest.py:33]: Do the new test files contain such cases of multiple other names so these changes are tested?

**Review by morcuended — APPROVED** [review]
Looks, good. Just some general questions.

The database should then be kept updated with these other names whenever they arise, right?

**Comment by juanep97** [comment]
> Looks, good. Just some general questions.
> 
> The database should then be kept updated with these other names whenever they arise, right?

Yes, there are currently 2 sources that have not been reduced at least in some epochs because the name was not identified, at least some nights. It does not usually happen, thou. One case it was because the name was wrong, the other because they just used a different name. Usually the regex we use is good enough.

**Comment by juanep97** [comment]
To be taken into account: when applying this update, we will probably need to manually indicate that there is a field rename in the migration.

**Review by morcuended — APPROVED** [review]
good, thanks

**Pull request merged**